omnipaxos/src/sequence_paxos/mod.rs

```diff
@@ -214,20 +214,21 @@ where
     pub(crate) fn resend_message_timeout(&mut self) {
         match &self.state {
             (Role::Leader, phase) => {
-                // Resend AcceptStopSign
+                // Resend AcceptStopSign or StopSign's decide
                 if *phase == Phase::Accept {
-                    // TODO: This is slow. Get stopsign from cache instead.
                     if let Some(ss) = self.internal_storage.get_stopsign() {
                         let decided_idx = self.internal_storage.get_decided_idx();
-                        if !ss.decided(decided_idx) {
-                            for follower in self.leader_state.get_promised_followers() {
-                                // resend if the follower has not accepted the stopsign
-                                if !ss.decided(self.leader_state.get_accepted_idx(follower)) {
-                                    self.send_accept_stopsign(follower, ss.stopsign.clone(), true);
-                                }
+
+                        for follower in self.leader_state.get_promised_followers() {
+                            if !ss.decided(decided_idx)
+                                && !ss.decided(self.leader_state.get_accepted_idx(follower))
```

Comment:
Not 100% sure what you mean by losing last N accepts. But i think either the next message received is then either a prepare (we will resync and get missed accepts) or an acceptDecide/decide/acceptStopSign and the sequence break will be discovered.

Reply:
When we create an empty stopsign:
- node A disconnects from cluster
- cluster proposes and decides a SS
- node reconnects to cluster
- leader creates AccSync with suffix=[ ], decided_idx=1 a stopsign and because the leader sees the it has a higher decided idx, it creates a snapshot of the empty log (can't snapshot a SS).
- on receiving the AccSync now Node A has a messed up log because it takes the SS and also the snapshot representing and empty log

Reply:
> Not 100% sure what you mean by losing last N accepts. But i think either the next message received is then either a prepare (we will resync and get missed accepts) or an acceptDecide/decide/acceptStopSign (the sequence break will be discovered).

Consider we send N Accept messages and then send a StopSign and they all get dropped. Since a StopSign is pending, the leader will not propose any further entries. Therefore we will never decide any of these unless we periodically send some message so that we can detect something has been dropped.

Reply:
but maybe we can solve that in a future PR

Reply:
on second thought, this is fine with your changes of resending stopsigns. But let's say we don't propose SS and don't propose any new entries after those N entries. Then those N entries will not be decided